docs/how-to-setup-a-validator-with-nginx-reverse-proxy.md

```diff
@@ -0,0 +1,176 @@
+---
+id: maintain-guides-how-to-setup-a-Validator-with-NGINX-reverse-proxy
+title: Set Up a Validator with NGINX Reverse Proxy
+sidebar_label: Set Up a Validator with NGINX Reverse Proxy
+---
+
+This guide assumes you have already the hardware ready with the appropriate specs. It has same configuration of the
+[polkadot secure validator](https://github.com/w3f/polkadot-secure-validator).
+
+In this guide, we will walk you through how to configure a reverse proxy in front of your validator. The
+validator uses the NGINX reverse proxy only to talk to the public facing nodes to protect it from certain types of attacks.
+
+### 1. Firewall configuration
+
+We will configure the firewall with ufw. There needs to be three main ports for this setup. The SSH port commonly is 22 which must be allowed, the proxy port which also must be allowed and the p2p port which in turn must be denied at the firewall level. In this example we will assign the port number `2435` to the proxy port and the port number `30333` to the p2p port.
+
+```bash
+# ssh port
+ufw allow 22/tcp
+# proxy port
+ufw allow 2435/tcp
+# libp2p port
+ufw deny 30333/tcp
+ufw enable
+# double check the firewall rules
+ufw verbose
+```
+
+### 2. Configure journald logs
+
+Create a file called `journald.conf` file inside the `/etc/systemd/` directory with the following content:
+
+```bash
+[Journal]
+Storage=persistent
+RateLimitIntervalSec=30s
+RateLimitBurst=20000
+SystemMaxUse=50G
+SystemMaxFileSize=512M
+SystemMaxFiles=100
+```
+
+Check out the [example journald configuration file](https://github.com/w3f/polkadot-secure-validator/blob/master/ansible/roles/polkadot-validator/files/journald.conf) for more available options.
+
+finally, run the following command to restart the journald service.
+
+```bash
+systemctl restart systemd-journald
+```
+
+### 3. NGINX reverse proxy setup
+
+First install NGINX with the following command.
+
+```
+apt install nginx
+```
+
+Next, create an NGINX configuration file called `nginx.conf` file inside the `/etc/nginx/` directory with the following content.
+
+```bash
+user www-data www-data;
+
+load_module /usr/lib/nginx/modules/ngx_stream_module.so;
+
+stream {
+  include streams-enabled/*;
+}
+
+http {
+  include sites-enabled/*;
+}
+
+events{
+}
+```
+
+This will import and make use of the [NGINX stream module](https://nginx.org/en/docs/stream/ngx_stream_core_module.html). In a nutshell, this module will allow continous streaming of data in or out of the validator machine with all the benefits of having an optimized reverse proxy.
```

Comment:
```suggestion
This will import and make use of the [NGINX stream module](https://nginx.org/en/docs/stream/ngx_stream_core_module.html). In a nutshell, this module allows for continuous streaming of data in or out of the validator machine with all the benefits of having an optimized reverse proxy.
```